driver/limits.py

```diff
@@ -80,7 +80,7 @@ def get_time_limit(component_limit, overall_limit):
         except NotImplementedError:
             returncodes.exit_with_driver_unsupported_error(CANNOT_LIMIT_TIME_MSG)
         else:
-            remaining_time = max(0, overall_limit - elapsed_time)
+            remaining_time = max(0, int(overall_limit - elapsed_time))
```

Comment:
I don't like the name `remaining_time` for something that isn't remaining time, but remaining time rounded down to the nearest second. (I'm fine with the `max` -- if the number is negative, I think it makes sense to argue that there is no remaining time rather than negative remaining time.) One option:

move the `int` conversion to the line `limit = remaining_time`: `limit = int(remaining_time)`. The same comment as below applies to a possible epsilon addition.

Reply:
I added a function to do the rounding (with an epsilon of 1ms) and called it when setting the limit.